knack/commands.py

```diff
@@ -124,7 +124,12 @@ def update_argument(self, param_name, argtype):
         # that coincides with the default
         if isinstance(arg_default, str):
             arg_default = DefaultStr(arg_default)
-        elif isinstance(arg_default, int):
+        elif type(arg_default) is int:  # pylint: disable=unidiomatic-typecheck
```

Comment:
Why not state it like this, it's much clear
```suggestion
        elif isinstance(arg_default, int) and not isinstance(arg_default, bool):
```